cgp/local_search/evolution_strategies.py

```diff
@@ -0,0 +1,218 @@
+
+        Parameters
+        ----------
+        individual : Individual
+            Individual for which to perform local search.
+        objective : Callable[[Callable[[np.ndarray[float]], np.ndarray[float]]], float]
+            Objective function used for the local search. Needs to accept
+            an numpy-compatible function as first argument (e.g., from
+            IndividualBase.to_numpy()) and return a float representing the
+            fitness of the individual.
+        seed : int
+            Seed for internal random number generator.
+        learning_rate_mu : float, optional
+            Learning rate for mean of search distribution.
+            Defaults to 1.0.
+        learning_rate_sigma : float, optional
+            Learning rate for standard deviation of search
+            distribution. Defaults to (3 + log(dim)) / (5.0 *
+            sqrt(dim)), where dim is the dimension of the search space.
+        population_size : int, optional
+            Number of samples evaluated per generation. Defaults to 4
+            + floor(3 * log(dim))), where dim is the dimension of the
+            search space..
+        max_generations : int, optional
+            Maximal number of generations. Defaults to 10.
+        min_sigma : float, optional
+            Minimal value for standard deviation of search
+            distribution. Defaults to 1e-6.
+        fitness_shaping : bool, optional
+            Whether to use fitness shaping. Defaults to True.
+        mirrored_sampling : bool, optional
+            Whether to use mirrored sampling. WARNING: Doubles the number
+            of samples per generation. Defaults to True.
+        n_processes : int, optional
+            Number of parallel processes. Defaults to 1.
+
+        """
+        self.objective = objective
+        self.learning_rate_mu = learning_rate_mu
+        self.learning_rate_sigma = learning_rate_sigma
+        self.population_size = population_size
+        self.max_generations = max_generations
+        self.min_sigma = min_sigma
+        self.fitness_shaping = fitness_shaping
+        self.mirrored_sampling = mirrored_sampling
+        self.n_processes = n_processes
+
+        self.rng = np.random.RandomState(seed)
+
+        self.sigma: Dict[int, np.ndarray[float]] = {}
+
+    def __call__(self, ind: "IndividualBase") -> None:
+
+        process_pool: Union[None, mp.pool.Pool]
+        if self.n_processes > 1:
+            process_pool = mp.Pool(processes=self.n_processes)
+        else:
+            process_pool = None
+
+        mu: np.ndarray[float]
+        params_names: List[str]
+        mu, params_names = ind.parameters_to_numpy_array(only_active_nodes=True)
+
+        sigma: np.ndarray[float]
+        if ind.idx in self.sigma:  # try loading sigma
+            sigma = self.sigma[ind.idx]
+            assert len(sigma) == len(mu)
+        elif ind.parent_idx in self.sigma and len(mu) == len(
+            self.sigma[ind.parent_idx]
```

Comment:
Probably I am getting something wrong, but what if lengths are equal to the parent, but its different active genes?

Reply:
that would indeed be a problem that's not handled well here. any idea how to avoid this?

Reply:
#907e2b6 fixes this issue